Source/missiles.cpp

```diff
@@ -4354,95 +4379,97 @@ void MI_Element(int i)
+		if (missile._mirange == 0) {
+			missile._mimfnum = 0;
 			SetMissAnim(i, MFILE_BIGEXP);
-			Missiles[i]._mirange = Missiles[i]._miAnimLen - 1;
-			Missiles[i].position.StopMissile();
+			missile._mirange = missile._miAnimLen - 1;
+			missile.position.StopMissile();
 		}
 	}
 	PutMissile(i);
 }
 
 void MI_Bonespirit(int i)
 {
-	Missiles[i]._mirange--;
-	int dam = Missiles[i]._midam;
-	int id = Missiles[i]._misource;
-	if (Missiles[i]._mimfnum == DIR_OMNI) {
-		ChangeLight(Missiles[i]._mlid, Missiles[i].position.tile, Missiles[i]._miAnimFrame);
-		if (Missiles[i]._mirange == 0) {
-			Missiles[i]._miDelFlag = true;
-			AddUnLight(Missiles[i]._mlid);
+	auto &missile = Missiles[i];
+	missile._mirange--;
+	int dam = missile._midam;
+	int id = missile._misource;
+	if (missile._mimfnum == DIR_OMNI) {
+		ChangeLight(missile._mlid, missile.position.tile, missile._miAnimFrame);
+		if (missile._mirange == 0) {
+			missile._miDelFlag = true;
+			AddUnLight(missile._mlid);
 		}
 		PutMissile(i);
 	} else {
-		Missiles[i].position.traveled += Missiles[i].position.velocity;
+		missile.position.traveled += missile.position.velocity;
 		UpdateMissilePos(i);
-		Point c = Missiles[i].position.tile;
+		Point c = missile.position.tile;
 		CheckMissileCol(i, dam, dam, false, c, false);
-		if (Missiles[i]._miVar3 == 0 && c == Point { Missiles[i]._miVar4, Missiles[i]._miVar5 })
-			Missiles[i]._miVar3 = 1;
-		if (Missiles[i]._miVar3 == 1) {
-			Missiles[i]._miVar3 = 2;
-			Missiles[i]._mirange = 255;
+		if (missile._miVar3 == 0 && c == Point { missile._miVar4, missile._miVar5 })
+			missile._miVar3 = 1;
+		if (missile._miVar3 == 1) {
+			missile._miVar3 = 2;
+			missile._mirange = 255;
 			auto *monster = FindClosest(c, 19);
 			if (monster != nullptr) {
-				Missiles[i]._midam = monster->_mhitpoints >> 7;
+				missile._midam = monster->_mhitpoints >> 7;
 				SetMissDir(i, GetDirection(c, monster->position.tile));
-				UpdateMissileVelocity(i, c, monster->position.tile, 16);
+				UpdateMissileVelocity(missile, c, monster->position.tile, 16);
 			} else {
 				Direction sd = Players[id]._pdir;
 				SetMissDir(i, sd);
-				UpdateMissileVelocity(i, c, c + sd, 16);
+				UpdateMissileVelocity(missile, c, c + sd, 16);
 			}
 		}
-		if (c != Point { Missiles[i]._miVar1, Missiles[i]._miVar2 }) {
-			Missiles[i]._miVar1 = c.x;
-			Missiles[i]._miVar2 = c.y;
-			ChangeLight(Missiles[i]._mlid, c, 8);
+		if (c != Point { missile._miVar1, missile._miVar2 }) {
+			missile._miVar1 = c.x;
+			missile._miVar2 = c.y;
+			ChangeLight(missile._mlid, c, 8);
 		}
-		if (Missiles[i]._mirange == 0) {
+		if (missile._mirange == 0) {
 			SetMissDir(i, DIR_OMNI);
-			Missiles[i].position.velocity = {};
-			Missiles[i]._mirange = 7;
+			missile.position.velocity = {};
+			missile._mirange = 7;
 		}
 		PutMissile(i);
 	}
 }
 
 void MI_ResurrectBeam(int i)
 {
+	auto &missile = Missiles[i];
 	Missiles[i]._mirange--;
 	if (Missiles[i]._mirange == 0)
 		Missiles[i]._miDelFlag = true;

```

Comment:
Looks like you ran out of steam here :)
```suggestion
	missile._mirange--;
	if (missile._mirange == 0)
		missile._miDelFlag = true;
```

Reply:
😄 updated. Thank you!